### ISSUE TITLE
 waybar is broken?

### ISSUE THREAD

**Issue by colemickens** [nix-community/nixpkgs-wayland#446]
Bit confused, `waybar` is broken for me on a local build. But I see an update went through?

```
Run-time dependency libudev found: YES 255
Run-time dependency libevdev found: YES 1.13.1
Run-time dependency libmpdclient found: YES 2.22
Run-time dependency xkbregistry found: YES 1.5.0
Run-time dependency jack found: YES 1.9.22
Run-time dependency wireplumber-0.4 found: NO (tried pkgconfig)

meson.build:95:17: ERROR: Dependency "wireplumber-0.4" not found, tried pkgconfig

A full log can be found at /build/source/build/meson-logs/meson-log.txt
```

A bit odd. The only scenario I can quickly think of is .... if nixos-unstable briefly got wireplumber and then later removed it?

Otherwise, idk, maybe it's conference+flight brain, but I'm not following.

**Comment by Artturin** [comment]
Already checked and we have to wait for a channel update to get a nixpkgs pr to make waybar use a older wireplumber

**Comment by colemickens** [comment]
Do you know how/why this landed then? https://github.com/nix-community/nixpkgs-wayland/commit/7d053c33b7a130ddada43ba09b089244390b3a23

**Comment by Artturin** [comment]
> Do you know how/why this landed then? [7d053c3](https://github.com/nix-community/nixpkgs-wayland/commit/7d053c33b7a130ddada43ba09b089244390b3a23)
> 
> (Sorry if I'm missing something obvious, it's probably time to close the computer.)

Issue is coming from wireplumber being updated to 0.5 in nixpkgs while waybar wants 0.4

You're using a different nixpkgs rev than the one pinned here

**Comment by colemickens** [comment]
Oh, I actually semi-checked for that. I could've sworn meson gave a more informative error in such cases...

Anyway, thanks for explaining.

**Comment by equals03** [comment]
> You're using a different nixpkgs rev than the one pinned here

this is probably not the right forum to ask, but I thought I would give it a whirl anyways as I have hit this problem too..  

if the inputs' (obviously in this case nixpkgs-wayland) nixpkgs input is "locked" to a specific version of nixpkgs that works for the build and my config is not specifically forcing it to follow the host, why would it try to use the nixpkgs from the host?

I completely understand and accept I am missing something fundamental in my understanding but I thought that was the whole point:
Regardless of the current version of nixpkgs on my host I should be able to build a package from another flake locked to its working version of nixpkgs as during the build as it provides the inputs to that flake/package?

Feel free to ignore this if off topic, but I'm hoping someone will answer as I have been looking for this answer for a while now (unsuccessfully) and I think its the last thing I need to reason about to fill out my understanding of nixos and its build system with flakes :smile:

**Comment by Artturin** [comment]
> > You're using a different nixpkgs rev than the one pinned here
> 
> this is probably not the right forum to ask, but I thought I would give it a whirl anyways as I have hit this problem too..
> 
> if the inputs' (obviously in this case nixpkgs-wayland) nixpkgs input is "locked" to a specific version of nixpkgs that works for the build and my config is not specifically forcing it to follow the host, why would it try to use the nixpkgs from the host?
> 
> I completely understand and accept I am missing something fundamental in my understanding but I thought that was the whole point: Regardless of the current version of nixpkgs on my host I should be able to build a package from another flake locked to its working version of nixpkgs as during the build as it provides the inputs to that flake/package?
> 
> Feel free to ignore this if off topic, but I'm hoping someone will answer as I have been looking for this answer for a while now (unsuccessfully) and I think its the last thing I need to reason about to fill out my understanding of nixos and its build system with flakes 😄

The overlay doesn't use the nixpkgs locked here, overlaying adds the packages to your nixpkgs.

**Comment by colemickens** [comment]
now that the nixpkgs change has landed in `nixos-unstable`, waybar has merged the 0.5 PR.

So we're probably going to need to re-fix this again in nixpkgs-wayland ahead of nixpkgs adjusting.
